libs/ledger-live-common/src/wallet-api/types.ts

```diff
@@ -62,3 +62,20 @@ export type AppManifest = {
 export type WalletAPISignedTransaction = SignedOperation;
 
 export type WalletAPISupportedCurrency = CryptoCurrency | TokenCurrency;
+
+export type GetWalletAPITransactionSignFlowInfos<
+  WalletAPITransaction,
+  LiveTransaction
+> = (tx: WalletAPITransaction) => {
+  canEditFees: boolean;
+  hasFeesProvided: boolean;
+  liveTx: Partial<LiveTransaction>;
+};
+
+export type AreFeesProvided<WalletAPITransaction> = (
+  tx: WalletAPITransaction
+) => boolean;
```

Comment:
I find it a bit misleading to use the same name for a generic and a real type, `WalletAPITransaction` is exported just above and in this lines `WalletAPITransaction` is a generic type and not the exported one.

Maybe we could simplify it, I have 2 ideas in mind:
```suggestion
export type AreFeesProvided<Transaction> = (
  tx: Transaction
) => boolean;
```
or
```suggestion
export type AreFeesProvided<T> = (
  tx: T
) => boolean;
```

And if you decide to change that we should do the same for `ConvertToLiveTransaction` and `GetWalletAPITransactionSignFlowInfos`

Reply:
Fixes done here -> 0ec3e458ab6b359714579a87ef4e471258e52b45

Rename + use of `extends` 👌